sunpy/io/special/genx.py

```diff
@@ -125,7 +126,7 @@ def unpack_string(self):
 
     def unpack_list(self, unpack_item):
         list = []
-        while 1:
+        while True:
             x = self.unpack_uint()
             if x == 0:
                 break
```

Comment:
This appears to be a prime candidate for the walrus operator!
```suggestion
        while (x := self.unpack_uint()) != 0:
```

Reply:
Added.